### PR TITLE
refactor(uncrustify): enable formatting for regexp and indent files

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/regexp.c [hunk 1]
```diff
@@ -322,20 +330,20 @@ static int reg_strict;          // "[abc" is illegal
...
 //  P        S     U  V  W  X     Z  [           _
-    1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1,
+  1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1,
 //     a     c  d     f     h  i     k  l  m  n  o
-    0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1,
+  0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1,
 //  p        s     u  v  w  x     z  {  |     ~
-    1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1
+  1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1
 };
```
Comment (zeertzjq) [inline src/nvim/regexp.c:347]: keep

File: src/nvim/indent_c.c [hunk 2]
```diff
@@ -3599,26 +3712,32 @@ int get_c_indent(void)
...
-   */
+
+  // NOTE: The '' around the backslashes in the following example shouldn't
+  // really exist, this is just a workaround to prevent the -Wcomment compiler
+  // warning.
+  //
+  // add extra indent if the previous line ended in a backslash:
+  //          "asdfasdf'\'
+  //              here";
+  //        char *foo = "asdf'\'
```
Comment (zeertzjq) [inline src/nvim/indent_c.c:3723]:
```suggestion
  //          "asdfasdf{backslash}
  //              here";
  //        char *foo = "asdf{backslash}
```

File: src/nvim/indent_c.c [hunk 3]
```diff
@@ -445,26 +444,27 @@ static int cin_isinit(void)
...
 
 /*
  * Recognize a switch label: "case .*:" or "default:".
  */
-bool cin_iscase(
-    const char_u *s,
-    bool strict     // Allow relaxed check of case statement for JS
-)
+bool cin_iscase(const char_u *s, bool strict     // Allow relaxed check of case statement for JS
+                )
```
Comment (mhinz) [inline src/nvim/indent_c.c:467]:
```
/// Recognize a switch label: "case .*:" or "default:".
///
/// @param s       blabla
/// @param strict  Allow relaxed check of case statement for JS
/// @return        moer blablalba
```

File: src/nvim/indent_c.c [hunk 4]
```diff
@@ -823,52 +830,54 @@ static int cin_islinecomment(const char_u *p)
...
+/// '}'.
+/// Don't consider "} else" a terminated line.
+/// If a line begins with an "else", only consider it terminated if no unmatched
+/// opening braces follow (handle "else { foo();" correctly).
+///
+/// @param incl_open   include '{' at the end as terminator
+/// @param incl_comma  recognize a trailing comma
+///
+/// @return  the character terminating the line (ending char's have precedence if
+///           both apply in order to determine initializations).
```
Comment (mhinz) [inline src/nvim/indent_c.c:843]: One space too much. ;]

File: src/nvim/indent_c.c [hunk 5]
```diff
@@ -1489,7 +1520,7 @@ static pos_T * find_match_char(char_u c, int ind_maxparen)
       curwin->w_cursor = *trypos;
       if ((trypos_wk = ind_find_start_CORS(NULL)) != NULL) {  // XXX
         ind_maxp_wk = ind_maxparen - (int)(cursor_save.lnum
-            - trypos_wk->lnum);
+                                           - trypos_wk->lnum);
```
Comment (mhinz) [inline src/nvim/indent_c.c:1523]: Might fit on the previous line.

File: src/nvim/indent_c.c [hunk 6]
```diff
@@ -1738,57 +1770,97 @@ void parse_cino(buf_T *buf)
...
-    case '#': buf->b_ind_hash_comment = n; break;
-    case 'N': buf->b_ind_cpp_namespace = n; break;
-    case 'k': buf->b_ind_if_for_while = n; break;
-    case 'E': buf->b_ind_cpp_extern_c = n; break;
-    case 'P': buf->b_ind_pragma = n; break;
-    }
-    if (*p == ',')
-      ++p;
+    case '>':
+      buf->b_ind_level = n; break;
```
Comment (mhinz) [inline src/nvim/indent_c.c:1787]: This is a bit weird. If we break up the oneliners, then `break` should get its own line as well, IMHO.

File: src/nvim/indent_c.c [hunk 7]
```diff
@@ -1908,13 +1980,11 @@ int get_c_indent(void)
     }
   }
 
-  /*
-   * Is it a non-case label?	Then that goes at the left margin too unless:
-   *  - JS flag is set.
-   *  - 'L' item has a positive value.
-   */
+  // Is it a non-case label?    Then that goes at the left margin too unless:
```
Comment (mhinz) [inline src/nvim/indent_c.c:1983]: Useless spaces between sentences.

File: src/nvim/indent_c.c [hunk 8]
```diff
@@ -1908,13 +1980,11 @@ int get_c_indent(void)
...
+  //  - JS flag is set.
+  //  - 'L' item has a positive value.
   if (original_line_islabel && !curbuf->b_ind_js
-           && curbuf->b_ind_jump_label < 0) {
+      && curbuf->b_ind_jump_label < 0) {
```
Comment (mhinz) [inline src/nvim/indent_c.c:1987]: Might fit on previous line.

File: src/nvim/indent_c.c [hunk 9]
```diff
@@ -2045,18 +2117,14 @@ int get_c_indent(void)
      * asterisk in the comment opener; otherwise, line up
      * with the first character of the comment text.
      */
-    if (done)
-      ;
-    else if (theline[0] == '*')
+    if (done) {} else if (theline[0] == '*') {
```
Comment (mhinz) [inline src/nvim/indent_c.c:2119]:
```c
if (done) {
} else if ... {
  ...
} else {
```

File: src/nvim/indent_c.c [hunk 10]
```diff
@@ -2672,25 +2764,26 @@ int get_c_indent(void)
...
-            /*
-             *	case xx:
-             *	    c = 99 +	    <- this indent plus continuation
-             **->	   here;
-             */
+            //  case xx:
+            //      c = 99 +        <- this indent plus continuation
+            // ->          here;
             if (lookfor == LOOKFOR_UNTERM
                 || lookfor == LOOKFOR_ENUM_OR_INIT) {
```
Comment (mhinz) [inline src/nvim/indent_c.c:2780]: Join with previous line

File: src/nvim/indent_c.c [hunk 11]
```diff
@@ -3401,14 +3507,14 @@ int get_c_indent(void)
...
-          && cin_isfuncdecl(NULL, cur_curpos.lnum + 1,
-              cur_curpos.lnum + 1)
-          && !cin_isterminated(theline, false, true)) {
+      && !cin_nocode(theline)
+      && vim_strchr(theline, '{') == NULL
+      && vim_strchr(theline, '}') == NULL
+      && !cin_ends_in(theline, (char_u *)":", NULL)
+      && !cin_ends_in(theline, (char_u *)",", NULL)
+      && cin_isfuncdecl(NULL, cur_curpos.lnum + 1,
+                        cur_curpos.lnum + 1)
```
Comment (mhinz) [inline src/nvim/indent_c.c:3515]: Prev lien?

**Review by mhinz — APPROVED** [review]

**Comment by justinmk** [comment]
do we have a `.git-blame-ignore-revs` PR ?

**Comment by dundargoc** [comment]
https://github.com/neovim/neovim/pull/18579

**Pull request merged**